include/aspect/newton.h

```diff
@@ -122,26 +123,53 @@ namespace aspect
       void parse_parameters (ParameterHandler &prm);
 
       /**
-       * A scaling factor for those terms of the Newton matrix that
-       * result from the linearization of the viscosity.
+       * A scaling factor used for scaling the
+       * derivative part of the Newton Stokes solver in the assembly.
        *
-       * See the get_newton_derivative_scaling_factor() function for an
-       * explanation of the purpose of this factor.
+       * The exact Newton matrix consists of the Stokes matrix plus a term
+       * that results from the linearization of the material coefficients.
+       * The scaling factor multiplies these additional terms. In a full
+       * Newton method, it would be equal to one, but it can be chosen
+       * smaller in cases where the resulting linear system has undesirable
+       * properties.
+       *
+       * If the scaling factor is zero, the resulting matrix is simply the
+       * Stokes matrix, and the resulting scheme is a defect correction
+       * (i.e., Picard iteration).
        */
       double              newton_derivative_scaling_factor;
 
       Stabilization       preconditioner_stabilization;
       Stabilization       velocity_block_stabilization;
 
+      /**
+       * Whether to use the Newton failsafe or not. If the failsafe is used, a failure
+       * of the linear solver is caught and we try to solve it again with both the
+       * preconditioner and the velocity block being stabilized with the SPD stabilization.
+       */
       bool                use_Newton_failsafe;
 
+      /**
+       * The nonlinear tolerance at which to switch the
+       * nonlinear solver from defect correction Picard to
+       * Newton.
+       */
       double              nonlinear_switch_tolerance;
+
       unsigned int        max_pre_newton_nonlinear_iterations;
       unsigned int        max_newton_line_search_iterations;
       bool                use_newton_residual_scaling_method;
       double              maximum_linear_stokes_solver_tolerance;
       double              SPD_safety_factor;
     };
+
+
+    /**
+     * Get a std::string describing the stabilization type used for the
+     * preconditioner.
+     */
+    std::string
+    to_string(const Newton::Parameters::Stabilization preconditioner_stabilization);
```

Comment:
The name look a bit too generic to me, because in the code you now get `Newton::to_string(some_parameter)`. I would prefer to call it something like `stabilization_to_string` or `stabilization_type_to_string`

Reply:
I chose that name because C++11 has a function of that name for all sorts of data types: 
http://www.cplusplus.com/reference/string/to_string/
http://en.cppreference.com/w/cpp/string/basic_string/to_string
So this is sort of a standard name to use in all contexts where you want to convert a something to a string, and the distinction is really only in the type of the argument.

Reply:
ah, oke. I didn't know that. Then I am fine with this change.